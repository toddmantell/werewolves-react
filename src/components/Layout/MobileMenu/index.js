import React from 'react';
import {arrayOf, object, string} from 'prop-types';
import styled from 'styled-components';

const Menu = styled.main`
	position: fixed;
	width: 100%;
	height: 50px;
`;

const MenuBar = ({menuItems}) => (
	<Menu>
		{
			menuItems.map(item => {
				<MenuItem {...item} />;
			})
		}
	</Menu>
);

MenuBar.propTypes = {
	menuItems: arrayOf(object)
};

export default MenuBar;