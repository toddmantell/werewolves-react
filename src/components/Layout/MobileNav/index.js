import React from "react"
import { arrayOf, object, string } from "prop-types"
import styled from "styled-components"

const Menu = styled.main`
  position: fixed;
  width: 100%;
  height: 50px;
`

const MobileNav = ({ menuItems }) =>
  <Menu>
    {menuItems.map(item => {
      ;<MenuItem {...item} />
    })}
  </Menu>

MobileNav.propTypes = {
  menuItems: arrayOf(object)
}

export default MobileNav
