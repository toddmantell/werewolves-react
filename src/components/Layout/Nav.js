import React from 'react';
import {Link} from 'react-router-dom';

const Nav = () => (
	<nav style={{backgroundColor: 'midnightblue'}}>
		<span style={{color: 'white', fontWeight: 600}}>
			<Link to="/moderator">Moderator</Link>
		</span>
		<span style={{color: 'white', fontWeight: 600}}>
			<Link to="/night">Night</Link>
		</span>
	</nav>
);

export default Nav;