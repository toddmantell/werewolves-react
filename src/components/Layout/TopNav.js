import React from "react"
import { Link } from "react-router-dom"
import PropTypes from "prop-types"

const TopNav = props =>
  <nav style={{ backgroundColor: "midnightblue" }}>
    <span style={{ color: "white", fontWeight: 600 }}>
      <Link to="/moderator">Moderator</Link>
    </span>
    <span style={{ color: "white", fontWeight: 600 }}>
      <Link to="/night">Night</Link>
    </span>
  </nav>

export default TopNav
