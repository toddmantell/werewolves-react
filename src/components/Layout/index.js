/* eslint curly: 0 */
import React from 'react';
import PropTypes from 'prop-types';
import TopNav from './Nav';
import Footer from './Footer';
import MobileNav from './MobileMenu';

const Layout = ({windowSize}) => (
	<div>
		{renderTopOrBottomNav(windowSize)}
		<h2>Header!</h2>
		<div style={{color: 'purple', fontWeight: '600'}}>
			Put app content here.
		</div>
		<Footer />
		{renderTopOrBottomNav(windowSize)}
	</div>
);

function renderTopOrBottomNav(windowSize) {
	if (windowSize > 400) return <TopNav />;
	else if (windowSize <= 400) return <MobileNav />;
	return null;
}

Layout.propTypes = {

};

export default Layout;