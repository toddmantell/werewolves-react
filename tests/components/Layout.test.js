import React from "react"
import Layout from "../../src/components/Layout"
import toJson from "enzyme-to-json"

describe("Name of the group", () => {
  it("should render the mobile nav when window size is < 400", () => {
    const shallowRenderedLayout = shallow(<Layout windowSize={350} />)

    expect(shallowRenderedLayout.find("MobileNav").length).toBe(1)
  })

  it("should render the desktop nav when window size is > 400", () => {
    const shallowRenderedLayout = shallow(<Layout windowSize={1200} />)

    expect(shallowRenderedLayout.find("TopNav").length).toBe(1)
  })

  it("should not render a nav when window size is null", () => {
    const shallowRenderedLayout = shallow(<Layout windowSize={undefined} />)

    expect(shallowRenderedLayout.find("Nav").length).toBe(0)
    expect(shallowRenderedLayout.find("MobileNav").length).toBe(0)
  })
})
