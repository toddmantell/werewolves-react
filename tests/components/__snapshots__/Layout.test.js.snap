// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Name of the group should render the desktop nav when window size is > 400 1`] = `
<div>
  <TopNav />
  <h2>
    Header!
  </h2>
  <div
    style={
      Object {
        "color": "purple",
        "fontWeight": "600",
      }
    }
  >
    Put app content here.
  </div>
  <Component />
</div>
`;

exports[`Name of the group should render the mobile nav when window size is < 400 1`] = `
<div>
  <h2>
    Header!
  </h2>
  <div
    style={
      Object {
        "color": "purple",
        "fontWeight": "600",
      }
    }
  >
    Put app content here.
  </div>
  <Component />
  <MenuBar />
</div>
`;
